src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java

```diff
@@ -86,21 +88,38 @@ public static void assertEnvelopeContainsOcrDataIfRequired(InputEnvelope envelop
      * @param pdfs     to assert against
      */
     public static void assertEnvelopeHasPdfs(InputEnvelope envelope, List<Pdf> pdfs) {
+        List<String> problems = new ArrayList<>();
+
+        List<String> duplicateFileNames =
+            envelope
+                .scannableItems
+                .stream()
+                .map(it -> it.fileName)
+                .collect(groupingBy(it -> it, counting()))
+                .entrySet()
+                .stream()
+                .filter(it -> it.getValue() > 1)
+                .map(it -> it.getKey())
+                .collect(toList());
```

Comment:
I've been looking into this for a while but only coming with either nasty hack or classic foreach loop with 2 sets figuring out what is duplicate what is not.

This streaming put me off.